### PR TITLE
Arming: Allow checking all EKF cores for health

### PR DESCRIPTION
**Pull request by WickedShell**
This is intended to actually check that all the EKF cores are actually healthy, rather then just having the primary core be healthy. I didn't tag the other vehicles yet, as it wasn't obvious to me under what situations Copter/Rover want to run the check.

Overall I'd love to see the AHRS arming check become common.

I haven't flight tested this yet, was just spun out as a side effect of Paul's presentation, and should be reviewed more.

### CODE REVIEW

**Comment by rmackay9** [comment]
I'm certainly happy for the check to be in rover and copter as well. I can't think of a reason we wouldn't want this in all vehicles..

File: libraries/AP_AHRS/AP_AHRS_NavEKF.cpp [hunk 1]
```diff
@@ -1114,6 +1114,30 @@ bool AP_AHRS_NavEKF::healthy(void) const
...
+        break;
+
+    case 2:
+    default:
+        if (_ekf2_started && EKF2.all_cores_healthy()) {
+            prearm_health = true;
+        }
+        break;
+
+    case 3:
```
Comment (peterbarker) [inline libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: Don't we need a `case 10`?  Better still, an enumeration so we don't miss cases?

Comment (WickedShell) [reply libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: argh, yes I forgot `case EKF_TYPE_SITL` when I yanked lines :)

**Comment by rmackay9** [comment]
I guess we're just waiting for @WickedShell to address the issue raised by @peterbarker above..

**Comment by WickedShell** [comment]
Still working on getting back from the conference...

**Comment by lvale** [comment]
@WickedShell Is it all EKF cores, or only the ones being used?

**Comment by proficnc** [comment]
I vote for all vehicles as well.

Thanks for adding this @WickedShell, its an important feature.

File: libraries/AP_AHRS/AP_AHRS_NavEKF.cpp (hunk 1 above)
Comment (WickedShell) [reply libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: Fixed. I yanked most of the switch down which is where the numbers came from in the first place :(

**Comment by rmackay9** [comment]
Looks like there's a compile error in Rover..

**Comment by WickedShell** [comment]
@lvale all the cores on the EKF you are using. (IE all EKF2 or all EKF3 cores)

**Pull request merged**